tests/Integration/Behaviour/Features/Context/Domain/Product/UpdateStockFeatureContext.php

```diff
@@ -0,0 +1,325 @@
+        $this->assertStringProperty($productForEditing, $data, 'pack_stock_type');
+        $this->assertIntegerProperty($productForEditing, $data, 'out_of_stock_type');
+        $this->assertIntegerProperty($productForEditing, $data, 'quantity');
+        $this->assertIntegerProperty($productForEditing, $data, 'minimal_quantity');
+        $this->assertStringProperty($productForEditing, $data, 'location');
+        $this->assertIntegerProperty($productForEditing, $data, 'low_stock_threshold');
+        $this->assertBoolProperty($productForEditing, $data, 'low_stock_alert');
+        $this->assertDateProperty($productForEditing, $data, 'available_date');
+
+        // Assertions checking isset() can hide some errors if it doesn't find array key,
+        // to make sure all provided fields were checked we need to unset every asserted field
+        // and finally, if provided data is not empty, it means there are some unnasserted values left
+        Assert::assertEmpty($data, sprintf('Some provided product stock fields haven\'t been asserted: %s', var_export($data, true)));
+    }
+
+    /**
+     * @Then product :productReference last stock movement has following details:
+     */
+    public function assertProductLastStockMovement(string $productReference, TableNode $table)
+    {
+        $movementData = $table->getRowsHash();
+        $productId = $this->getSharedStorage()->get($productReference);
+
+        /** @var StockMovementRepository $stockMovementRepository */
+        $stockMovementRepository = $this->getContainer()->get('prestashop.core.api.stock_movement.repository');
+        $params = new QueryStockMovementParamsCollection();
+        $params->fromArray([
+            'productId' => $productId,
+        ]);
+        $movements = $stockMovementRepository->getData($params);
+        if (count($movements) <= 0) {
+            throw new RuntimeException(sprintf('No stock movement found for product %s', $productReference));
+        }
+
+        $lastMovement = $movements[0];
+        foreach ($movementData as $movementField => $movementValue) {
+            Assert::assertEquals(
+                $movementValue,
+                $lastMovement[$movementField],
+                sprintf(
+                    'Invalid stock movement field %s, expected %s instead of %s',
+                    $movementField,
+                    $movementValue,
+                    $lastMovement[$movementField]
+                )
+            );
+        }
+    }
+
+    /**
+     * @Then product :productReference has no stock movements
+     */
+    public function assertProductHasNoStockMovement(string $productReference)
+    {
+        $productId = $this->getSharedStorage()->get($productReference);
+
+        /** @var StockMovementRepository $stockMovementRepository */
+        $stockMovementRepository = $this->getContainer()->get('prestashop.core.api.stock_movement.repository');
```

Comment:
I think you can initialize this in constructor

Reply:
I prefer to call it "on demand" as this context might be used in other features but without using this assessment so no need to get the service then
And getting a service multiple times has no performance drawbacks